semgrep-core/synthesizing/Synthesizer.mli

```diff
@@ -0,0 +1,2 @@
+(* range as "start row:start col-end row:end col" -> filename -> (label * pattern) list *)
+val synthesize_patterns: string -> string -> (string * string) list
```

Comment:
Ideally this would use better type than string, something like
synthesize_patterns: Range.t -> Common.filename -> (string * Pattern.t) list

then the caller of that would use the parsing/unparsing functions to convert from or to string.